app/src/main/java/shoppinglist/com/shoppinglist/ShoppingListActivity.java

```diff
@@ -0,0 +1,179 @@
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationServices;
+
+import java.util.List;
+
+import shoppinglist.com.shoppinglist.camera.CameraActivity;
+import shoppinglist.com.shoppinglist.database.DatabaseHelper;
+import shoppinglist.com.shoppinglist.database.ShoppingListDatabase;
+import shoppinglist.com.shoppinglist.database.exceptions.PersistingFailedException;
+import shoppinglist.com.shoppinglist.database.orm.ShoppingItem;
+import shoppinglist.com.shoppinglist.database.orm.ShoppingList;
+import shoppinglist.com.shoppinglist.network.NetBlaster;
+
+public class ShoppingListActivity extends ActionBarActivity implements GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener {
+
+
+    ShoppingList shoppingList = null;
+    ShoppingListDatabase shoppingListDatabase = null;
+    private GoogleApiClient mGoogleApiClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_shopping_list);
+
+
+        //TODO remove singleton or make getInstance api cleaner
+        //this.deleteDatabase(DatabaseHelper.DATABASE_NAME);
+        shoppingListDatabase = DatabaseHelper.getInstance(getApplicationContext());
+
+        try {
+            shoppingList = shoppingListDatabase.getCurrentShoppingList();
+        } catch (PersistingFailedException e) {
+            Toast.makeText(this, "Couldn't get ShoppingList.", Toast.LENGTH_SHORT).show();
+        }
+
+        List<ShoppingItem> items = shoppingList.getItems();
+
+        final ShoppingListAdapter shoppingListAdapter = new ShoppingListAdapter(this, items, shoppingListDatabase);
+        final ListView shoppingListView = (ListView)findViewById(R.id.shopping_list);
+        shoppingListView.setAdapter(shoppingListAdapter);
+
+
+        Button clearButton = (Button)findViewById(R.id.clear_button);
+        clearButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                TextView inputText = (TextView) findViewById(R.id.shopping_input);
+                inputText.setText("");
+                try {
+                    shoppingListDatabase.clearList(shoppingList);
+                }
+                catch(PersistingFailedException e){
+                    Toast.makeText(ShoppingListActivity.this, "Clearing List failed.", Toast.LENGTH_SHORT).show();
+                }
+                shoppingListAdapter.clear();
+                shoppingListAdapter.notifyDataSetChanged();
```

Comment:
Does the data set actually change in the event of a failed clear? If not, then it might make sense to move this into the try block. Also depends on if you want the data to be cleared from the adapter even though it may not have been successfully cleared from the database.